### PR TITLE
fixing Operator/Authority choice in line

### PR DESCRIPTION
**Pull request by ue71603**
like #613  #341 but against master

### CODE REVIEW

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 1]
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
...
 		<xsd:annotation>
-			<xsd:documentation>Property Elements of a LINE.</xsd:documentation>
+			<xsd:documentation>Property elements of a LINE.</xsd:documentation>
 		</xsd:annotation>
 		<xsd:sequence>
-			<xsd:choice minOccurs="0">
+			<xsd:element ref="AuthorityRef" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>Primary OPERATOR of LINE.</xsd:documentation>
+					<xsd:documentation>A line can only have no Authority when this information is provided in a ResponsibilitySet.</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:414]: I dislike this documenation sentense.

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 2]
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
...
 				</xsd:annotation>
-				<xsd:element ref="AuthorityRef"/>
-				<xsd:element ref="OperatorRef"/>
-			</xsd:choice>
+			</xsd:element>
+			<xsd:element ref="OperatorRef" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Operators should be set, even when the same as the Authority. Can be omitted when provided via ResponsibilitySet</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:419]: No, don't allow it to be omitted...

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 3]
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
...
+				</xsd:annotation>
+			</xsd:element>
 			<xsd:element name="additionalOperators" type="transportOrganisationRefs_RelStructure" minOccurs="0">
```
Comment (skinkie) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:422]: I think additionalOperators where added when we had a choice of Authority or Operator. Should we discuss this too?

**Comment by ue71603** [comment]
@skinkie  I think one still can have additional operators.  The main case is 1-1, but there are others. Or do you want to change it?

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd (hunk 1 above)
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:414]:
```suggestion
					<xsd:documentation>A line should alway have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
```

Comment (trurlurl) [reply xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:414]:
```suggestion
					<xsd:documentation>A line should always have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
```

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd (hunk 2 above)
Comment (ue71603) [reply xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd]: changed

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd (hunk 3 above)
Comment (ue71603) [reply xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:422]: my comment ok?

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd (hunk 1 above)
Comment (ue71603) [reply xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd]: updated

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 4]
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
...
+			<xsd:element ref="AuthorityRef" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>Primary OPERATOR of LINE.</xsd:documentation>
+					<xsd:documentation>A line should alway have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
 				</xsd:annotation>
-				<xsd:element ref="AuthorityRef"/>
-				<xsd:element ref="OperatorRef"/>
-			</xsd:choice>
+			</xsd:element>
+			<xsd:element ref="OperatorRef" minOccurs="0">
```
Comment (ue71603) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:417]:
```suggestion
```

**Comment by ue71603** [comment]
fixed a bug I introduced
@skinkie  @Aurige  @duex 
it is currently possible to provide neither Authorithy nor Operator (and do everything in ResponsibilitySets).
If you don't want that, then intervene (btw this is what SBB is doing)

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 5]
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
...
 		<xsd:annotation>
-			<xsd:documentation>Property Elements of a LINE.</xsd:documentation>
+			<xsd:documentation>Property elements of a LINE.</xsd:documentation>
 		</xsd:annotation>
 		<xsd:sequence>
-			<xsd:choice minOccurs="0">
+			<xsd:element ref="AuthorityRef" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>Primary OPERATOR of LINE.</xsd:documentation>
+					<xsd:documentation>A line should alway have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
```
Comment (Robbendebiene) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:414]:
```suggestion
					<xsd:documentation>A line should always have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
```

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 6]
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
...
 		<xsd:annotation>
-			<xsd:documentation>Property Elements of a LINE.</xsd:documentation>
+			<xsd:documentation>Property elements of a LINE.</xsd:documentation>
 		</xsd:annotation>
 		<xsd:sequence>
-			<xsd:choice minOccurs="0">
+			<xsd:element ref="AuthorityRef" minOccurs="0">
 				<xsd:annotation>
-					<xsd:documentation>Primary OPERATOR of LINE.</xsd:documentation>
+					<xsd:documentation>A line should always have an Authority, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:414]: Do you want to write AUTHORITY here or isn't that a Transmodel concept?

Comment (ue71603) [reply xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:414]: changed

File: xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd [hunk 7]
```diff
@@ -406,16 +406,19 @@ Rail transport, Roads and Road transport
...
-					<xsd:documentation>Primary OPERATOR of LINE.</xsd:documentation>
+					<xsd:documentation>A line should always have an AUTHORITY, even when the information is provided in a ResponsibilitySet.</xsd:documentation>
 				</xsd:annotation>
-				<xsd:element ref="AuthorityRef"/>
-				<xsd:element ref="OperatorRef"/>
-			</xsd:choice>
+			</xsd:element>
+			<xsd:element ref="OperatorRef" minOccurs="0">
+				<xsd:annotation>
+					<xsd:documentation>Operators should be set, even when the same as the Authority. In some cases Operator/Authority are managed through ResponsibilitySet. However, for compatibility OperatorRef and AuthorityRef still should be filled in.</xsd:documentation>
```
Comment (skinkie) [inline xsd/netex_part_1/part1_networkDescription/netex_line_version.xsd:419]:
```suggestion
					<xsd:documentation>An OPERATOR should be set, even when the same as the AUTHORITY. In some cases OPERATOR or AUTHORITY are managed through a ResponsibilitySet. However, for compatibility OperatorRef and AuthorityRef still should be filled in.</xsd:documentation>
```

**Review by skinkie — APPROVED** [review]

**Review by trurlurl — APPROVED** [review]

**Review by Aurige — APPROVED** [review]

**Pull request merged**